.github/actions/setup-python-build-env/action.yml

```diff
@@ -0,0 +1,11 @@
+name: "Setup Build Environment"
+description: "Install everything needed to build"
+
+runs:
+  using: "composite"
+  steps:
+  - name: Upgrade pip and install poetry
+    shell: bash
+    run: |
+      pip install --upgrade pip
```

Comment:
Can you also pin `pip` to a specific version? I remember some of our builds broke over night when pip changed the way it resolves dependencies.